src/api_manager/context/request_context.cc

```diff
@@ -292,6 +301,64 @@ void RequestContext::FillHttpHeaders(const Response *response,
   }
 }
 
+void RequestContext::FillJwtPayloads(service_control::ReportRequestInfo *info) {
+  auto serverConfig = service_context_->config()->server_config();
+  if (serverConfig->has_service_control_config() &&
+      serverConfig->service_control_config().log_jwt_payload().size() != 0) {
+    JwtCache &jwt_cache = service_context()->jwt_cache();
+    JwtCache::ScopedLookup lookup(&jwt_cache, auth_token_);
```

Comment:
It is odd to have a cache lookup.

Reply:
right, should use auth_claims_ directly. Done.